docs/spec/dataclasses.rst

```diff
@@ -526,6 +529,52 @@ This includes, but is not limited to, the following semantics:
   dataclass must be explicitly annotated as e.g. ``x: ClassVar[Final[int]] =
   3``.
 
+Converters
+^^^^^^^^^^
+
+The ``converter`` parameter can be specified in a field definition to provide a callable
+used to convert values when assigning to the associated attribute. This feature allows for automatic type
+conversion and validation during attribute assignment.
+
+Converter behavior:
+
+* The converter is used for all attribute assignment, including assignment
+  of default values, assignment in synthesized ``__init__`` methods
+  and direct attribute setting (e.g., ``obj.attr = value``).
+* The converter is not used when reading attributes, as the attributes should already have been converted.
+
+Typing rules for converters:
+
+* The ``converter`` must be a callable that must accept a single positional argument
+  (but may optionally accept other arguments, which are ignored for typing purposes).
```

Comment:
Provided the other arguments are optional (i.e., they have a default or are variadic). Hopefully that's clear but maybe we can use a wording that makes it more explicit.